File: Tutorial with code with harry/rotation.java
public class rotation {
    
}
#include <iostream>
#include <vector>

using namespace std;

vector<int> leftRotate(const vector<int>& arr, int d) {
    int n = arr.size();
    vector<int> rotatedArr(n);

    // Calculate the index from where rotation starts
    int rotationIndex = d % n;

    // Copy elements after rotation index
    for (int i = 0; i < n - rotationIndex; ++i) {
        rotatedArr[i] = arr[i + rotationIndex];
    }

    // Copy elements before rotation index
    for (int i = 0; i < rotationIndex; ++i) {
        rotatedArr[n - rotationIndex + i] = arr[i];
    }

    return rotatedArr;
}

int main() {
    // Example usage
    vector<int> arr = {1, 2, 3, 4, 5};
    int d = 2; // Number of left rotations

    vector<int> rotatedArr = leftRotate(arr, d);

    cout << "Array after " << d << " left rotations: ";
    for (int num : rotatedArr) {
        cout << num << " ";
    }
    cout << endl;

    return 0;
}

File: Tutorial with code with harry/rotation.cpp
#include <iostream>

using namespace std;

void leftRotate(int arr[], int d, int n)
{
    for (int i = 0; i < d; i++)
    {
        int temp = arr[0];
        for (int j = 0; j < n - 1; j++)
        {
            arr[j] = arr[j + 1];
        }
        arr[n - 1] = temp;
    }
}

int main()
{
    int arr[] = {1, 2, 3, 4, 5, 6, 7};
    int n = sizeof(arr) / sizeof(arr[0]);
    int d = 2;

    leftRotate(arr, d, n);
    
    cout << "rotated array: ";
    for (int i = 0; i < n; i++)
    {
        cout  << arr[i] << " ";
    }

    return 0;
}